src/Appwrite/Auth/OAuth2/Amazon.php

```diff
@@ -59,31 +64,27 @@ public function getLoginURL(): string
     /**
      * @param string $code
      *
-     * @return string
+     * @return array
      */
-    public function getAccessToken(string $code): string
+    public function getTokens(string $code): array
     {
-        $headers = ['Content-Type: application/x-www-form-urlencoded;charset=UTF-8'];
-        $accessToken = $this->request(
-            'POST',
-            'https://api.amazon.com/auth/o2/token',
-            $headers,
-            \http_build_query([
-                'code' => $code,
-                'client_id' => $this->appID ,
-                'client_secret' => $this->appSecret,
-                'redirect_uri' => $this->callback ,
-                'grant_type' => 'authorization_code'
-            ])
-        );
-        
-        $accessToken = \json_decode($accessToken, true);
-
-        if (isset($accessToken['access_token'])) {
-            return $accessToken['access_token'];
+        if(empty($this->tokens)) {
+            $headers = ['Content-Type: application/x-www-form-urlencoded;charset=UTF-8'];
+            $this->tokens = \json_decode($this->request(
```

Comment:
Just so I understand correctly, this stores the whole response from the access token request right? But we only provide ways to get the access token and refresh token publicly?

How does this handle nested JSON objects?

Reply:
Yes, methods `getRefreshToken` and `getAccessToken` are on parent - on OAuth2.php. I decided to put it there because all of our current providers follow the same response structure. If there is some provider that does not, the function could be overwritten.